tests/regression/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+        process::Command,
+    };
+
+    const COST: u64 = 1_000_000; //configurable threshold for regression
+
+    struct InstrumentationControl;
+
+    impl InstrumentationControl {
+        fn stop_instrumentation(&self) {
+            cg::cachegrind::stop_instrumentation();
+        }
+
+        fn start_instrumentation(&self) {
+            cg::cachegrind::start_instrumentation();
+        }
+    }
+    // environment variable to determine whether to run under valgrind or solely test functionality
+    fn is_running_under_valgrind() -> bool {
+        env::var("ENABLE_VALGRIND").is_ok()
+    }
+
+    fn valgrind_test<F>(test_name: &str, test_body: F) -> Result<(), s2n_tls::error::Error>
+    where
+        F: FnOnce(&InstrumentationControl) -> Result<(), s2n_tls::error::Error>,
+    {
+        if !is_running_under_valgrind() {
+            let ctrl = InstrumentationControl;
+            test_body(&ctrl)
+        } else {
+            run_valgrind_test(test_name);
+            Ok(())
+        }
+    }
+
+    //test to create new config, set security policy, host_callback information, load/trust certs, and build config
+    #[test]
+    fn test_set_config() {
+        valgrind_test("test_set_config", |ctrl| {
+            ctrl.stop_instrumentation();
+            ctrl.start_instrumentation();
+            let keypair_rsa = CertKeyPair::default();
+            let _config =
+                set_config(&security::DEFAULT_TLS13, keypair_rsa).expect("Failed to build config");
+            Ok(())
+        })
+        .unwrap();
+    }
+
+    #[test]
+    fn test_rsa_handshake() {
+        valgrind_test("test_rsa_handshake", |ctrl| {
+            ctrl.stop_instrumentation();
+            // Example usage with RSA keypair (default)
+            let keypair_rsa = CertKeyPair::default();
+            let config = set_config(&security::DEFAULT_TLS13, keypair_rsa)?;
+            // Create a pair (client + server) using that config, start handshake measurement
+            let mut pair = TestPair::from_config(&config);
+            // Assert a successful handshake
+            ctrl.start_instrumentation();
+            assert!(pair.handshake().is_ok());
+            ctrl.stop_instrumentation();
+            Ok(())
+        })
+        .unwrap();
+    }
+    // function to run specified test using valgrind
+    fn run_valgrind_test(test_name: &str) {
+        let exe_path = std::env::args().next().unwrap();
+        let _dir_path = create_dir(Path::new("target/cg_artifacts"));
+        let output_file = format!("target/cg_artifacts/cachegrind_{}.out", test_name);
+        let output_command = format!("--cachegrind-out-file={}", &output_file);
+        let mut command = Command::new("valgrind");
+        command
+            .args(["--tool=cachegrind", &output_command, &exe_path, test_name])
+            .env_remove("ENABLE_VALGRIND"); //ensures that the recursive call is made to the actual harness code block rather than back to this function
```

Comment:
```suggestion
            // ensures that the recursive call is made to the actual harness code block rather than back to this function
            .env_remove("ENABLE_VALGRIND");
```
This line is just getting a little long, let's go ahead and split it onto two.